semantic_model_generator/tests/yaml_to_semantic_model_test.py

```diff
@@ -0,0 +1,34 @@
+import os
+
+import pytest
+from strictyaml import YAMLValidationError
+
+from semantic_model_generator.data_processing.proto_utils import yaml_to_semantic_model
```

Comment:
If the proto definition changes, is there any way to catch that in these tests? 

Maybe we can create a set of expected fields based on the current proto definition and ensure that matches the current compiled proto spec.

Just want to make sure if we're updating the proto we're also updating the strictyaml verification.

Reply:
what do you mean? The way this is implemented we do DFS in the entire proto definition, so any schema changes would be automatically picked up

Reply:
got it - makes sense